src/NumericalAlgorithms/DiscontinuousGalerkin/NumericalFluxes/HartenLaxVanLeer.hpp

```diff
@@ -0,0 +1,216 @@
+ * \f$c_\text{max} = 0\f$ (i.e. all characteristics move in the same direction)
+ * the HLL flux reduces to pure upwinding.
+ *
+ * \anchor hll_ref [1] A. Harten, P. D. Lax, B. van Leer, On Upstream
+ * Differencing and Godunov-Type Schemes for Hyperbolic Conservation Laws,
+ * SIAM Rev. [25 (1983) 35](https://doi.org/10.1137/1025002)
+ *
+ * \anchor estimates_ref [2] S. F. Davis, Simplified Second-Order Godunov-Type
+ * Methods, SIAM J. Sci. Stat. Comput.
+ * [9 (1988) 445](https://doi.org/10.1137/0909030)
+ */
+template <typename System>
+struct HartenLaxVanLeer {
+ private:
+  using char_speeds_tag = typename System::char_speeds_tag;
+  using variables_tags = typename System::variables_tags;
+
+ public:
+  /// The minimum signal velocity bounding
+  /// the wavespeeds on one side of the interface.
+  struct MinSignalSpeed : db::SimpleTag {
+    using type = Scalar<DataVector>;
+    static std::string name() noexcept { return "MinSignalSpeed"; }
+  };
+
+  /// The maximum signal velocity bounding
+  /// the wavespeeds on one side of the interface.
+  struct MaxSignalSpeed : db::SimpleTag {
+    using type = Scalar<DataVector>;
+    static std::string name() noexcept { return "MaxSignalSpeed"; }
+  };
+
+  using package_tags =
+      tmpl::push_back<tmpl::append<db::split_tag<db::add_tag_prefix<
+                                       ::Tags::NormalDotFlux, variables_tags>>,
+                                   db::split_tag<variables_tags>>,
+                      MinSignalSpeed, MaxSignalSpeed>;
+
+  using argument_tags = tmpl::append<
+      db::split_tag<db::add_tag_prefix<::Tags::NormalDotFlux, variables_tags>>,
+      db::split_tag<variables_tags>, char_speeds_tag>;
+
+ private:
+  template <typename VariablesTagList, typename NormalDoFluxTagList>
+  struct package_data_helper;
+
+  template <typename... VariablesTags, typename... NormalDotFluxTags>
+  struct package_data_helper<tmpl::list<VariablesTags...>,
+                             tmpl::list<NormalDotFluxTags...>> {
+    static void function(
+        const gsl::not_null<Variables<package_tags>*> packaged_data,
+        const db::item_type<NormalDotFluxTags>&... n_dot_f_to_package,
+        const db::item_type<VariablesTags>&... u_to_package,
+        const db::item_type<char_speeds_tag>& characteristic_speeds) noexcept {
+      expand_pack((get<VariablesTags>(*packaged_data) = u_to_package)...);
+      expand_pack(
+          (get<NormalDotFluxTags>(*packaged_data) = n_dot_f_to_package)...);
+
+      get<MinSignalSpeed>(*packaged_data) = make_with_value<Scalar<DataVector>>(
+          characteristic_speeds[0],
+          std::numeric_limits<double>::signaling_NaN());
+      get<MaxSignalSpeed>(*packaged_data) = make_with_value<Scalar<DataVector>>(
+          characteristic_speeds[0],
+          std::numeric_limits<double>::signaling_NaN());
+
+      for (size_t s = 0; s < characteristic_speeds[0].size(); ++s) {
```

Comment:
I think here it would be good to have a comment along the lines of: `Find the min and max signal speed of all the variables at each grid point.`